app/controllers/users/registrations_controller.rb

```diff
@@ -29,10 +29,20 @@ def destroy
     end
   end
 
+  def password
+    redirect_to root_path if current_user.nil? 
+  end
+
   protected
 
-  def update_resource(resource, params)
+  def update_resource(resource, params) 
+    if params[:password_confirmation].nil? && resource.errors.size == 0
+      flash.now[:notice] = 'User succesfully updated!'
+    end
     resource.update(params)
+    if resource.errors.size == 0 && !params[:password_confirmation].nil?
+      flash.now[:notice] = 'Password succesfully updated!'
```

Comment:
Same as above, keep the "green" success message and delete the "red" success message.